Automattic/jetpack
# Jetpack: Start an initial full sync if the site is connected when Jetpack is activated

#### Changes proposed in this Pull Request:
* An initial full sync is started when the site is registered. If the site is already connected via another plugin when Jetpack is activated, a Sync process won't be started until a change is made on the site that triggers a Sync. So, let's start an initial full sync during plugin activation if the site is already connected.

#### Jetpack product discussion
* p91TBi-6EK-p2#comment-7444

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:
1. Install a plugin that uses the connection package but does not use the sync package, such as Jetpack Boost.
2. Connect the plugin to WPCOM.
3. Clear the `jetpack_sync_fill_status` option if it contains data: `wp option delete jetpack_sync_full_status`
4. Using your sandbox, verify that the value of the `jetpack_version` option for the site is false. If this site has previously been connected, you may need to delete the option:
    a. `switch_to_blog( {your blog id} );
    b. 'return delete_option( 'jetpack_version' );
5. Install the master branch of Jetpack but do not activate it.
6. Activate Jetpack using WPCLI: `wp plugin activate jetpack`
7. Using your sandbox, check the value of the `jetpack_version` option for the site. It should be false because Sync didn't send any data when Jetpack activated.
    a. `switch_to_blog( {your blog id} );
    b. 'return get_option( 'jetpack_version' );
8. Deactivate Jetpack.
9. Using your sandbox, verify that the value the of `jetpack_version` option is still false. If not, delete the option.
10. Checkout this branch of Jetpack, but do not activate it.
11. Activate Jetpack using WPCLI: `wp plugin activate jetpack`
12. Using your sandbox, check the value of the `jetpack_version` option for the site. It should be `10.5-a.0`.
